File: 1297-MaximumNumberOfBalloons/1297-MaximumNumberOfBalloons.py
# Last updated: 6/29/2026, 10:21:53 AM
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        mp = {}
        mp['b'] = 0
        mp['a'] = 1
        mp['l'] = 2
        mp['o'] = 3
        mp['n'] = 4
        
        count = [0] * 5

        for s in text:
            if s in mp:
                count[mp[s]] += 1
        
        count[mp['l']] //= 2
        count[mp['o']] //= 2

        return min(count)

File: 1297-MaximumNumberOfBalloons/1297-MaximumNumberOfBalloons.cpp
// Last updated: 6/29/2026, 10:21:59 AM
class Solution {
public:
    int maxNumberOfBalloons(string text) {
        unordered_map<char, int> idxmp;
        idxmp['b'] = 0;
        idxmp['a'] = 1;
        idxmp['l'] = 2;
        idxmp['o'] = 3;
        idxmp['n'] = 4;
        vector<int> count(5, 0);
        for(auto &c : text) {
            if(idxmp.count(c) == 0) continue;
            count[idxmp[c]]++;
        }

        count[3] /= 2;
        count[2] /= 2;
        return *min_element(count.begin(), count.end());
    }
};